lib/formatter.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const generateFunction = require('generate-function');
+
+function getFormatter(metric, defaultLabels) {
+	const defaultLabelNames = Object.keys(defaultLabels);
+	const name = metric.name;
+	const help = escapeString(metric.help);
+	const type = metric.type;
+	const labelsCode = getLabelsCode(metric, defaultLabelNames, defaultLabels);
+
+	const gen = generateFunction();
+	gen(`
+  function format(item, escapeLabelValue, getValueAsString) {
+    const info = '# HELP ${name} ${help}\\n# TYPE ${name} ${type}\\n';
+    let values = '';
+    for (const val of item.values || []) {
+      val.labels = val.labels || {};
+      let metricName = val.metricName || '${name}';
+      const labels = \`${labelsCode}\`;
+      const hasLabels = labels.length > 0;
+      metricName += \`\${hasLabels ? '{' : ''}\${labels}\${hasLabels ? '}' : ''}\`;
+      let line = \`\${metricName} \${getValueAsString(val.value)}\`;
+      values += \`\${line}\\n\`;
+    }
+    return info + values;
+  }`);
+	return gen.toFunction();
```

Comment:
You could save a dependency and just use the Function constructor directly, since you're not making use of any of `generateFunctions`'s other abilities.